src/Mapbender/PrintBundle/Component/ImageExportService.php

```diff
@@ -144,8 +138,157 @@ private function getImages()
+                    /** @var EventDispatcherInterface $eventDispatcher */
+                    $eventDispatcher = $this->container->get('event_dispatcher');
+                    $proxy = new WmsProxy($eventDispatcher, $proxyConfig, $proxyQuery, $this->getLogger());
+                    break;
+                default:
+                    $proxy = new CommonProxy($proxyConfig, $proxyQuery, $this->getLogger());
+                    break;
+            }
+            /** @var \Buzz\Message\Response $buzzResponse */
+            $buzzResponse = $proxy->handle();
+            $response = $this->convertBuzzResponse($buzzResponse);
+        }
+        return $response;
+    }
+
+    /**
+     * Convert a Buzz Response to a Symfony HttpFoundation Response
+     *
+     * @todo: This belongs in owsproxy; it's the only part of Mapbender that uses Buzz
+     *
+     * @param \Buzz\Message\Response $buzzResponse
+     * @return Response
+     */
+    public static function convertBuzzResponse($buzzResponse)
+    {
+        // adapt header formatting: Buzz uses a flat list of lines, HttpFoundation expects a name: value mapping
+        $headers = array();
+        foreach ($buzzResponse->getHeaders() as $headerLine) {
+            $parts = explode(':', $headerLine, 2);
+            if (count($parts) == 2) {
+                $headers[$parts[0]] = $parts[1];
+            }
+        }
+        $response = new Response($buzzResponse->getContent(), $buzzResponse->getStatusCode(), $headers);
+        $response->setProtocolVersion($buzzResponse->getProtocolVersion());
+        $statusText = $buzzResponse->getReasonPhrase();
+        if ($statusText) {
+            $response->setStatusCode($buzzResponse->getStatusCode(), $statusText);
+        }
+        return $response;
+    }
+
+    /**
+     * Converts a http response to a GD image, respecting the mimetype.
+     *
+     * @param string $storagePath for temp file storage
+     * @param Response $response
+     * @return resource|null GD image or null on failure
+     */
+    protected function serviceResponseToGdImage($storagePath, $response)
+    {
+        file_put_contents($storagePath, $response->getContent());
+        $contentType = trim($response->headers->get('content-type'));
+        switch ($contentType) {
+            case (preg_match("/image\/png/", $contentType) ? $contentType : !$contentType) :
+                return imagecreatefrompng($storagePath);
+                break;
+            case (preg_match("/image\/jpeg/", $contentType) ? $contentType : !$contentType) :
+                return imagecreatefromjpeg($storagePath);
+                break;
+            case (preg_match("/image\/gif/", $contentType) ? $contentType : !$contentType) :
+                return imagecreatefromgif($storagePath);
+                break;
+            default:
+                return null;
+                $this->getLogger()->debug("Unknown mimetype " . trim($response->headers->get('content-type')));
```

Comment:
Does this work? How can the logger be called if the function returns before? Or is this some php black magic?

Reply:
It doesn't work. Same as before. It's an equivalent transform of old behaviour.
See this commit: https://github.com/mapbender/mapbender/pull/711/commits/3056d755b89a07314441a9ccc1d9e4e88f6b8111#diff-aa7793858e9f53e396ac560ee2b6a9b0L103

I have already worked on further improvements, including the reenabling of that logging message, but they are beyond the scope of this pull.